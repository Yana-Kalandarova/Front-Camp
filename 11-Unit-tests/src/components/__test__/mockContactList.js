import mockContact from './mockContact';

const mockContactList = [
  mockContact,
  mockContact,
];

export default mockContactList;
