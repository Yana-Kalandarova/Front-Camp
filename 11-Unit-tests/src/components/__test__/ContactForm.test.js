import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import { shallow, mount } from 'enzyme';
import ContactForm from '../ContactForm';
import mockContactList from './mockContactList';

const props = {
  match: {
    params: {
      contactId: '1a2b3c',
    }
  }
}

describe('ContactForm', () => {
  it('renders correctly', () => {
    expect(
      mount(
        <BrowserRouter>
          <ContactForm contactList={mockContactList} onUpdateContact={jest.fn()} {...props} />
        </BrowserRouter>
      )
    ).toMatchSnapshot()
  });
});
