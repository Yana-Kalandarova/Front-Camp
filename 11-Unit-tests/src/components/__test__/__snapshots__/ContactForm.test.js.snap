// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactForm renders correctly 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <withRouter(ContactForm)
      contactList={
        Array [
          Object {
            "_id": "1a2b3c",
            "name": "Test Name",
            "phoneNumber": "12345",
          },
          Object {
            "_id": "1a2b3c",
            "name": "Test Name",
            "phoneNumber": "12345",
          },
        ]
      }
      match={
        Object {
          "params": Object {
            "contactId": "1a2b3c",
          },
        }
      }
      onUpdateContact={[MockFunction]}
    >
      <ContactForm
        contactList={
          Array [
            Object {
              "_id": "1a2b3c",
              "name": "Test Name",
              "phoneNumber": "12345",
            },
            Object {
              "_id": "1a2b3c",
              "name": "Test Name",
              "phoneNumber": "12345",
            },
          ]
        }
        history={
          Object {
            "action": "POP",
            "block": [Function],
            "createHref": [Function],
            "go": [Function],
            "goBack": [Function],
            "goForward": [Function],
            "length": 1,
            "listen": [Function],
            "location": Object {
              "hash": "",
              "pathname": "/",
              "search": "",
              "state": undefined,
            },
            "push": [Function],
            "replace": [Function],
          }
        }
        location={
          Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          }
        }
        match={
          Object {
            "isExact": true,
            "params": Object {},
            "path": "/",
            "url": "/",
          }
        }
        onUpdateContact={[MockFunction]}
      >
        <form
          className="contact_form"
          onSubmit={[Function]}
        >
          <div
            className="contact_form-field-wrap"
          >
            <label
              htmlFor="name"
            >
              Name:
            </label>
            <input
              className="contact_form-field"
              id="name"
              name="name"
              onChange={[Function]}
              placeholder="Name"
              type="text"
              value=""
            />
            <ErrorMessage
              message=""
            />
          </div>
          <div
            className="contact_form-field-wrap"
          >
            <label
              htmlFor="phoneNumber"
            >
              Phone Number:
            </label>
            <input
              className="contact_form-field"
              id="phoneNumber"
              name="phoneNumber"
              onChange={[Function]}
              placeholder="Phone"
              type="tel"
              value=""
            />
            <ErrorMessage
              message=""
            />
          </div>
          <div
            className="contact_btn-wrap"
          >
            <Link
              className="contact_btn-cancel"
              to="/"
            >
              <LinkAnchor
                className="contact_btn-cancel"
                href="/"
                navigate={[Function]}
              >
                <a
                  className="contact_btn-cancel"
                  href="/"
                  onClick={[Function]}
                >
                  Cancel
                </a>
              </LinkAnchor>
            </Link>
            <button
              className="contact_btn-save"
              disabled={true}
              type="submit"
            >
              Save
            </button>
          </div>
        </form>
      </ContactForm>
    </withRouter(ContactForm)>
  </Router>
</BrowserRouter>
`;
